Wikia/selenium-tests
# Soc 1677

Test case 1: for logged in user:
- click upvote button and verify it adds an upvote
- click upvote button again and verify it removes the upvote

Test case 2: for anonymous user: 
- click upvote button 
- verify it does not count an upvote

Vote area / resolution | Desktop | Mobile 
 ---|---|---
post details vote area | TC1 :ok: TC2 :ok: | TC1 :ok: TC2 :ok:
post details reply vote area |TC1 :ok: TC2 :ok: | TC1 :ok: TC2 :ok:
post list vote area | TC1 :ok: TC2 :ok: | TC1 :ok: TC2 :ok:

Why Upvote functionality was moved to separate test classes ?
- PostDetailsUpvote contains 4 tests and the test class is 141 lines of code long. 
Thus adding it to the PostDetails file (71 lines) would unnecessary increase the file length to above 200 lines.
Long files are less readable and harder to navigate through. 
- PostListUpvote contains 8 tests and the test class is 83 lines of code long. 
Thus adding it to the PostList file (189 lines) would unnecessary increase the file length to above 250 lines.
Long files are less readable and harder to navigate through. 

Share Functionality

Test Case 1:
click sharing icon and verify that proper sharing option are displayed

Test Case 2:
verify that proper sharing option are displayed by default

area / resolution | Desktop | Mobile 
 ---|---|---
post details | :zero:  | TC2 :ok:
post list  | TC1 :ok:  | TC1 :ok:
